src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -123,20 +123,37 @@ public static boolean containsAnnotation(DetailAST ast, Set<String> annotations)
 
         if (!annotations.isEmpty()) {
             final DetailAST firstMatchingAnnotation = findFirstAnnotation(ast, annotationNode -> {
-                DetailAST identNode = annotationNode.findFirstToken(TokenTypes.IDENT);
-                if (identNode == null) {
-                    identNode = annotationNode.findFirstToken(TokenTypes.DOT)
-                            .findFirstToken(TokenTypes.IDENT);
-                }
-
-                return annotations.contains(identNode.getText());
+                return doesAnnotationMatch(annotationNode, annotations);
             });
             result = firstMatchingAnnotation != null;
         }
 
         return result;
     }
 
+    /**
+     * Checks if annotation is found in the set of provided annotations.
+     *
+     * @param annotations A collection of annotations to look through.
+     * @param annotationNode Annotation to look for.
+     * @return {@code true} if the annotation is contained in the set.
+     */
+    private static boolean doesAnnotationMatch(DetailAST annotationNode, Set<String> annotations) {
```

Comment:
Make function getAnnotatioFullIdent
Checking for contains should be outside of new function

Reply:
Done!